staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -100,6 +101,12 @@ func (c *dynamicClient) NamespacedSubresource(resource schema.GroupVersionResour
 	return &dynamicResourceClient{client: c, resource: resource, namespace: namespace, subresource: subresource}
 }
 
+func (c *dynamicResourceClient) Namespace(ns string) DynamicResourceInterface {
```

Comment:
this makes the dynamic client non-threadsafe... it means if I run several workers off the same dynamic client, they mutate the client and can pick up namespace values intended for other calls

Reply:
hmm... actually, it's doing a shallow copy, nevermind.